File: modules/odr_core/odr_core/enums.py
# SPDX-License-Identifier: Apache-2.0
from enum import Enum


class ContentType(str, Enum):
    IMAGE = "image"
    VIDEO = "video"
    VOICE = "voice"
    MUSIC = "music"
    TEXT = "text"


class ContentStatus(str, Enum):
    PENDING = "pending"
    AVAILABLE = "available"
    UNAVAILABLE = "unavailable"
    DELISTED = "delisted"


class ContentSourceType(str, Enum):
    URL = "url"
    PATH = "path"
    HUGGING_FACE = "hugging_face"


class ReportStatus(str, Enum):
    PENDING = "pending"
    REVIEWED = "reviewed"
    RESOLVED = "resolved"


class AnnotationSourceType(str, Enum):
    CONTENT_DESCRIPTION = "content_description"
    SPATIAL_ANALYSIS = "spatial_analysis"
    TAGS = "tags"
    OTHER = "other"


class ReportType(str, Enum):
    ILLEGAL_CONTENT = "illegal_content"
    MALICIOUS_ANNOTATION = "malicious_annotation"
    OTHER = "other"


class EmbeddingEngineType(str, Enum):
    IMAGE = "image"
    VIDEO = "video"
    VOICE = "voice"
    MUSIC = "music"
    TEXT = "text"

File: modules/odr_frontend/src/db/schemas/enums.ts
/* SPDX-License-Identifier: Apache-2.0 */
import { pgEnum } from 'drizzle-orm/pg-core';

export const contentsourcetype = pgEnum('contentsourcetype', ['URL', 'PATH', 'HUGGING_FACE']);
export const contentstatus = pgEnum('contentstatus', [
	'PENDING',
	'AVAILABLE',
	'UNAVAILABLE',
	'DELISTED'
]);
export const contenttype = pgEnum('contenttype', ['IMAGE', 'VIDEO', 'VOICE', 'MUSIC', 'TEXT']);
export const embeddingenginetype = pgEnum('embeddingenginetype', [
	'IMAGE',
	'VIDEO',
	'VOICE',
	'MUSIC',
	'TEXT'
]);
export const reportstatus = pgEnum('reportstatus', ['PENDING', 'REVIEWED', 'RESOLVED']);
export const usertype = pgEnum('usertype', ['user', 'bot']);
